src/btree/multi.ts

```diff
@@ -0,0 +1,78 @@
+import { RangeResolver } from "../resolver";
+import { MemoryPointer } from "./node";
+
+export class LinkedMetaPage {
+	private resolver: RangeResolver;
+	private offset: number;
+
+	constructor(resolver: RangeResolver, offset: number) {
+		this.resolver = resolver;
+		this.offset = offset;
+	}
+
+	async root(): Promise<MemoryPointer | null> {
+		// we seek by 12 bytes since offset is 8 bytes, length is 4 bytes
+		const { data } = await this.resolver({
+			start: this.offset,
+			end: this.offset + 12,
+		});
+
+		if (data.byteLength !== 12) {
+			return null;
+		}
+
+		const view = new DataView(data);
+
+		const pointerOffset = view.getBigUint64(0);
+		const lengthOffset = view.getUint32(8);
+
+		return {
+			offset: pointerOffset,
+			length: lengthOffset,
+		};
+	}
+
+	async metadata(): Promise<ArrayBuffer> {
+		const { data: lengthData } = await this.resolver({
+			start: this.offset + 24,
+			end: this.offset + 28,
+		});
+
+		const lengthView = new DataView(lengthData);
+
+		// read the first four because that represnts length
+		const metadataLength = lengthView.getUint32(0);
+
+		// reard from 4 to 4 + length for metadata
+		const { data: metadata } = await this.resolver({
+			start: this.offset + 28,
+			end: this.offset + 28 + metadataLength,
+		});
+
+		return metadata;
+	}
+
+	async next(): Promise<LinkedMetaPage | null> {
+		const { data } = await this.resolver({
+			start: this.offset + 12,
+			end: this.offset + 12 + 8,
+		});
+
+		if (data.byteLength !== 8) {
+			return null;
+		}
+
+		const view = new DataView(data);
```

Comment:
I used `DataView` as this was frequently used in the previous index file version.

I followed the same pattern, retrieve the ArrayBuffer data from the range resolver, assert the byte length, then feed it into the data view.